### DISCUSSION TITLE
Pagination confusion

### DISCUSSION THREAD

**Discussion by thedangler** [saloonphp/saloon#380] [Q&A]
I have an api endpoint that does implement pagination like limi/offset.
I've connected that api to the a UI and I want the UI to control the next page.
I'm looking at the documentation and it looks like saloon automatically gets every page for you?

How would I go about telling it what page to get?

**Comment by Sammyjo20** [comment]
Hey @thedangler I'm really sorry for missing this discussion question! 

If it's still pertinent - the OffsetPaginator is what you'd need: https://docs.saloon.dev/installable-plugins/pagination/limit-offset-pagination

I would recommend the above page though if you haven't already. By default we assume there are two query parameters to be sent: `limit` and `offset` however some APIs like to use `top` and `skip`. You can customise this on the paginator class.

https://docs.saloon.dev/installable-plugins/pagination/limit-offset-pagination#assumptions-made-with-the-offsetpaginator

**Reply by thedangler** [reply]
Hi Sammy,
No worries this is an ongoing project.
I have the pagination implemented but the docs loop through each page. Is it possible to use pagination but doing $request->next() or $request->page(3) ?

Thanks
